docs/custom_readers.md

```diff
@@ -0,0 +1,147 @@
+(custom-readers)=
+
+# Custom Readers
+
+This page explains how to write a custom reader for VirtualiZarr, to extract chunk references from an archival data format not already supported by the main package.
+This is advanced material intended for 3rd-party developers, and assumes you have read the page on [Data Structures](data_structures.md).
+
+## What is a VirtualiZarr reader?
+
+All VirtualiZarr readers are simply functions that accept a path to a file of a specific format, and return an instance of the `ManifestStore` class containing information about the contents of that file.
+
+```python
+from virtualizarr.manifests import ManifestStore
+
+def a_custom_reader(path: str, **kwargs) -> ManifestStore:
+    # parse the file contents here
+
+    # construct the Manifeststore
+    manifest_store = ManifestStore(...)
+
+    return manifest_store
+```
+
+## What is the responsibility of a reader?
+
+The VirtualiZarr package really does four separate things.
+In order, it:
+
+1. Maps the contents of common archival file formats to the Zarr data model, including references to the locations of the chunks.
+2. Loads chosen variables into memory (the `loadable_variables`).
+3. Provides a way to combine arrays of chunk references using a convenient API (the Xarray API).
+3. Allows persisting these references to storage for later use, in either the Kerchunk or Icechunk format.
+
+**VirtualiZarr readers are responsible for the entirety of step (1).**
+In other words, all of the assumptions required to map the data model of an archival file format to the Zarr data model, and the logic for doing so for a specific file, together constitute a reader.
+
+This design provides a neat separation of concerns, which is helpful in two ways:
+1. The Xarray data model is subtly different from the Zarr data model (see below), so as the final objective is to create a virtual store which programmatically maps Zarr API calls to the archival file format at read-time, it is useful to separate that logic up front, before we convert to use the xarray virtual dataset representation and potentially subtly confuse matters.
+2. It also allows us to support reading data from the file via the `ManifestStore` interface, without using Xarray directly.
+
+## Reading data from the `ManifestStore`
+
+As well as being a well-defined representation of the archival data in the Zarr model, you can also read chunk data directly from the `ManifestStore` object.
+
+This works because the `ManifestStore` class is an implementation of the Zarr-Python `zarr.abc.Store` interface, and uses the `obstore` package internally to actually fetch chunk data when requested.
```

Comment:
This section is missing a lot of detail about the store registry and credentials, but maybe you would like to add that in a follow-up PR @maxrjones ?

Reply:
I think I over-engineered the store registry and with the proposal in https://github.com/zarr-developers/VirtualiZarr/pull/568 it could be removed.